js-core/homeworks/homework-10/src/main.js

```diff
@@ -0,0 +1,221 @@
+
+console.log('\n\nЗадача 2\n');
+
+var ezjQuery = {
+  value : '',
+  add(node){
+    if(typeof node !== 'string'){
+      console.log('ERROR: value must be a string!');
+      return;
+    }
+
+    this.value += `<${node}></${node}>`;
+    console.log(this.value);
+    return this;
+  }
+};
+
+ezjQuery
+  .add('body') // <body></body>
+  .add('div') // <body></body><div></div>
+  .add('h1') // <body></body><div></div><h1></h1>
+  .add(11); // ERROR!
+
+/*
+ *
+ * TASK 3
+ * Доработйте метод add чтобы на каждом вызове следующий
+ * тэг помещался внутри предыдущего !
+*/
+/*
+ ---
+ * И добавьте объекту ezjQuery метод render, который будет возвращать
+ * сгенерированную строку
+ -----
+ * Методу add - второй параметр, который будет размещать
+ * информацию внутри тэга
+ *
+ */
+
+console.log('\n\nЗадача 3\n');
+var ezjQuery3 = {
+  value : [],
+
+  add(node,text){
+    if(typeof node !== 'string'){
+      console.log('ERROR: value must be a string!');
+      return;
+    }
+
+    let _text = text || '';
+    this.value.push([node,_text]);
+
+    console.log(this._bildString());
+    return this;
+  },
+
+  _bildString(){
+    return this.value.reduceRight(function(resString,node){
+        return `<${node[0]}>${node[1]}${resString}</${node[0]}>`;
+    },'')
+  },
+
+  render(){
+    let res = this._bildString();
+    this.value = [];
+    return res;
+  },
+
+  toString(){
+    return this._bildString();
+  }
+
+};
+
+
+
+
+
+// example
+
+var helloList = ezjQuery3
+  .add('body') // <body></body>
+  .add('div','block') // <body><div></div></body>
+  .add('ul','list') // <body><div><ul></ul></div></body>
+  .add('li', 'Hello') //<body><div><ul><li>Hello</li></ul></div></body>
+  // .render();
+;
+console.log('helloList.toString: ', helloList.toString()); // <body><div><ul><li>Hello</li></ul></div></body>
+helloList.render();
+//  Обратите внимание, что после вызова render создание строки началось сначала
+console.log('ВЫзвали render, сбросли строку');
+
+var bodyDiv = ezjQuery3
+  .add('body') //<body></body>
+  .add('div') //<body><div></div></body>
+  .render();
+
+console.log('bodyDiv.toString: ', bodyDiv.toString()); //<body><div></div></body>
+
+// Для выполнивших все задания
+// сделайте document.write(helloList) увидите результат :)
+document.write(helloList);
+console.log('\n\nЗадача SUPER\n');
+
+
+
+
+// @SUPER
+/*
+ * Переименуйте объект ezjQuery в $.
+ * Создание перевого метода должено быть без метода
+ *
+ * $('body').add('li', 'hi').render() // <body><li>hi</li></body>
+ *
+ * */
+ /*
+  немного непонятная постановка задачи, в плене сделать объект, но вызвать его как функцию.
+  Потому сделал не объект а функцию, которая возвращает даный нужный объект.
+ */
+
+function $(node){
+  let obj = {};
+  obj.value = [];
+
+  obj.add = function(node, text){
+
+    if(typeof node !== 'string'){
+      console.log('ERROR: value must be a string!');
+      return;
+    }
+
+    let _text = text || '';
+    this.value.push([node,_text]);
```

Comment:
data-structure choose good, but abstraction behind what every index mean would be pretty hard one day in future

maybe something like this would told about your code more

{tagName: node, tagContent: text}